blockdag/dag.go

```diff
@@ -1096,11 +1099,11 @@ func (node *blockNode) verifyAndBuildUTXO(dag *BlockDAG, transactions []*util.Tx
 		return nil, nil, nil, nil, err
 	}
 
-	diffFromTxs, err := node.diffFromTxs(pastUTXO, transactions)
-	if err != nil {
-		return nil, nil, nil, nil, err
+	pastUTXODiffSet, ok := pastUTXO.(*DiffUTXOSet)
```

Comment:
I don't really like the cast to *DiffUTXOSet, but it looks indeed necessary in the way you do things.
What was bad in the way things worked up until now?